unumpy/random/_multimethods.py

```diff
@@ -0,0 +1,389 @@
+import functools
+import operator
+from uarray import create_multimethod, mark_as, all_of_type, Dispatchable
+import builtins
+
+create_numpy = functools.partial(create_multimethod, domain="numpy.random")
+
+from .._multimethods import (
+    ClassOverrideMetaWithConstructor,
+    ClassOverrideMetaWithGetAttr,
+    ClassOverrideMetaWithConstructorAndGetAttr,
+    ndarray,
+    _identity_argreplacer,
+    _self_argreplacer,
+    _first2argreplacer,
+    _first3argreplacer,
+    mark_dtype,
+)
+
+
+@create_numpy(_identity_argreplacer)
+def default_rng(seed=None):
+    return ()
+
+
+class RandomState(metaclass=ClassOverrideMetaWithConstructor):
+    pass
```

Comment:
We should add methods for these classes as well.

Reply:
Not sure I understand what you mean.

Reply:
If you look at: https://github.com/Quansight-Labs/unumpy/blob/master/unumpy/_multimethods.py#L244-L295

You will see how the `ufunc` class defines methods. The same needs to be done here.

Reply:
Just for `RandomState` or `Generator` as well? As I understand we just have to write the multimethods again but as class methods, right?

Reply:
Prioritize `Generator`, but also `RandomState`.

Reply:
> You will see how the `ufunc` class defines methods. The same needs to be done here.

I tried calling `ufunc`'s `accumulate` method but it returns a `BackendNotImplementedError`. I think the methods are broken and need to be rewritten in terms of `overridden_class`.

Reply:
Does `reduce` work?

Reply:
Yes, it works if `ureduce` is mapped in `_implementations` in a given backend.

Reply:
Maybe we need a way to set the `domain` of classes to `numpy.classname`.

Reply:
I have a working implementation that serves as a simple solution (although maybe not the ideal) which maps the `Generator`'s methods. In the NumPy backend would be something like this:

```python
for k, v in unumpy.random.Generator.__dict__.items():
    if isinstance(v, _Function):
        _implementations[v] = getattr(np.random.Generator, k)
```

So for each multimethod in unumpy's `Generator` class it maps the corresponding method in NumPy's `Generator`. I think this `for` loop is at least better than manually adding the methods to the dictionary since they are manifold.

> Maybe we need a way to set the `domain` of classes to `numpy.classname`.

This sounds like a more interesting solution since the problem resides in the incomplete path to a given class method. I've looked into `__qualname__` which seems to give the full path so we might want to use something like that.